File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[][] a = new int[N][2];
        
        StringTokenizer st;
        for(int i=0; i<N; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            a[i][0] = Integer.parseInt(st.nextToken());
			a[i][1] = Integer.parseInt(st.nextToken());
        }
        
        Arrays.sort(a, new Comparator<int[]>() {		
	        @Override
	        public int compare(int[] b1, int[] b2) {
		        if(b1[0] == b2[0]) {		
			        return b1[1] - b2[1];
		        } else {
			        return b1[0] - b2[0];
		        }
    	    }
        });
        
        StringBuilder sb = new StringBuilder();
		for(int i=0; i<N; i++) {           
			sb.append(a[i][0] + " " + a[i][1]).append('\n');
		}
		System.out.println(sb);
            
    }
}

File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(const vector<int>& b1, const vector<int>& b2);

int main() {
    int N;
    cin >> N;
    vector<vector<int>> a(N, vector<int>(2));

    for (int i = 0; i < N; i++) {
        cin >> a[i][0] >> a[i][1];
    }

    sort(a.begin(), a.end(), compare);

    for (int i = 0; i < N; i++) {
        cout << a[i][0] << " " << a[i][1] << '\n';
    }

    return 0;
}

bool compare(const vector<int>& b1, const vector<int>& b2) {
    if (b1[0] == b2[0]) {
        return b1[1] < b2[1];
    } else {
        return b1[0] < b2[0];
    }
}